CI: Avoid path filtering in "required" workflows

OnkarRuikar opened the pull request:
~~The workflow checks only `files/en-us/_redirects.txt` file. [_It doesn't work on any other file_](https://github.com/mdn/yari/blob/65f1e4b793f5ce7fd0d4a4a0a2b623db67af336d/tool/cli.ts#L257C51-L257C65).~~

~~It is not required to run this workflow in every PR. This PR reduces the workflows scope to only the `_redirects.txt` file and the workflow file.~~

~~Also, there is no point in making this workflow "required" in settings.~~

Edit: Had to revert the original changes due to [the edge case](https://github.com/mdn/content/pull/28277#issuecomment-1659904324). Now the PR focuses only on avoiding path filtering.

/cc @bsmth

bsmth reviewed (approved):
Thanks, @OnkarRuikar - good cleanup, leaving a +1 👍🏻 

>there is no point in making this workflow "required" in settings.

Why not? It seems like one of the more useful jobs to have PRs block if the author made a tooling mistake that we don't want to land on main:

https://github.com/mdn/content/actions/runs/5387620912/job/14582323137

OnkarRuikar commented:
>> there is no point in making this workflow "required" in settings.
>
> Why not? It seems like one of the more useful jobs to have PRs block if the author made a tooling mistake that we don't want to land on main:

True! We do need these as required. Another (following) issue was lingering in my mind that is why the statement.

I was also thinking of the issue of these "required" workflows not running on other files and disabling the merge button. In other words we don't want to run redirects-check command for all the PRs but we do want to run the workflow as it is "required". GitHub docs suggests [not to use path filter for "required" workflows](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks). So updating these workflows using [dorny/paths-filter](https://github.com/dorny/paths-filter) action as requested by @Josh-Cena.

yin1999 commented:
Hi Onkar, in translated-content, we have occured the following case:

There is a redirection from `some-old-slug` to `new-slug`, but the contributor just created a document and the slug is `some-old-slug`, when we merge this PR, the redirection rule is broken. But in this case, we have merge one PR that have moved documents (by [`sync-translated-content`](https://github.com/mdn/translated-content/blob/main/.github/workflows/sync-translated-content.yml)). I'm not sure is there any edge case may break the redirection rule.

OnkarRuikar commented on .github/workflows/pr-check_redirects_file.yml:
```diff
@@ -0,0 +1,42 @@
+        with:
+          filters: |
+            required_files :
+              - "files/en-us/_redirects.txt"
```
```suggestion
              - ".nvmrc"
              - "files/**"
```
Handle the [edge case](https://github.com/mdn/content/pull/28277#issuecomment-1658653879)

OnkarRuikar committed:
Avoid path filering for 'requried' workflows

OnkarRuikar committed:
Update .github/workflows/pr-test.yml

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>

OnkarRuikar commented:
> but the contributor just created a document and the slug is `some-old-slug`, 

This is a very rare case but it is possible so putting the old "files/**" back.

So the the PRs scope has now reduced to only avoiding path filtering for "required" workflows. I've changed the PR title and updated the code to do just that.

bsmth reviewed (approved):
Thanks @OnkarRuikar - I think we're good to go
